src/cloud-api-adaptor/docker/README.md

```diff
@@ -80,9 +78,68 @@ For changing the CAA image to your custom built image (eg. `quay.io/myuser/cloud
 you can use the following:
 
 ```bash
-kubectl set image ds/cloud-api-adaptor-daemonset -n confidential-containers-system cloud-api-adaptor-con=quay.io/myuser/cloud-api-adaptor
+export CAA_IMAGE=quay.io/myuser/cloud-api-adaptor
+kubectl set image ds/cloud-api-adaptor-daemonset -n confidential-containers-system cloud-api-adaptor-con="$CAA_IMAGE"
 ```
 
+## Running the CAA e2e tests
+
+### Test Prerequisites
+
+Following prerequisites are needed on the test system:
+
+- make
+- go
+- yq
+- kubectl
+- kind
+- docker
+  
+A `prereqs.sh` helper script is available under `src/cloud-api-adaptor/docker` to install/uninstall the prerequisites.
+
+Note: If using the `prereqs.sh` helper script to install the prerequisites, then reload the shell to ensure new permissions
+are in place to run `docker` and other commands.
+
+### Test Execution
+
+In order to run the tests, edit the file `src/cloud-api-adaptor/test/provisioner/docker/provision_docker.properties`
+and update the `CAA_IMAGE` and `CAA_IMAGE_TAG` variables with your custom CAA image and tag.
+
+You can run the CAA e2e [tests/e2e/README.md](../test/e2e/README.md) by running the following command:
+
+```sh
+TEST_PODVM_IMAGE=<podvm-image> TEST_PROVISION=yes CLOUD_PROVIDER=docker TEST_PROVISION_FILE=$(pwd)/test/provisioner/docker/provision_docker.properties test-e2e
+```
+
+This will create a two node kind cluster, automatically download the pod VM image mentioned in the `provision_docker.properties`
+file and run the tests.
+
+Note: To overcome docker rate limiting issue or to download images from private registries,
```

Comment:
I hit this issue with the kind pods, so set up the config.json as described, but then hit a similar issue in the guest pull:
```
Error: failed to create containerd task: failed to create shim task: failed to pull manifest Registry error: url https://index.docker.io/v2/library/nginx/manifests/latest, envelope: OCI API errors: [OCI API error: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit]: unknown
```
Maybe we should consider trying to switch away from docker.io based images in our test code if there is a different mirror of nginx available? Kata uses `quay.io/sjenning/nginx:1.15-alpine`, but that is pretty old

Reply:
Using docker images will most likely always hit rate limit for e2e due to multiple runs. Using your personal login might help but again it depends on which plan you have. 
Overall switching to images in either quay or github itself is a better alternative for reliable testing.

Reply:
FYI - I've created https://github.com/confidential-containers/cloud-api-adaptor/pull/1900 to try and help with this